### PR TITLE
WFLY-17048 Remove javax.persistence persistence providers

### PR DESCRIPTION
**Pull request by scottmarlow**
Signed-off-by: Scott Marlow <smarlow@redhat.com>
https://issues.redhat.com/browse/WFLY-17048

### CODE REVIEW

**Comment by scottmarlow** [comment]
I'll build and test again locally to see if I missed something [that might cause](https://ci.wildfly.org/viewLog.html?buildId=328260&buildTypeId=WF_PullRequest_LinuxJdk17):
```
Caused by: java.lang.IllegalArgumentException: WFLYLNCHR0003: Invalid directory, could not find 'jboss-modules.jar' in '/opt/buildAgent/work/e34a6f994de9f7c6/testsuite/integration/ws/target/wildfly'
	at org.wildfly.core.launcher.Environment.validateWildFlyDir(Environment.java:239)
	at org.wildfly.core.launcher.Environment.validateWildFlyDir(Environment.java:227)
	at org.wildfly.core.launcher.StandaloneCommandBuilder.of(StandaloneCommandBuilder.java:99)
	at org.jboss.as.arquillian.container.managed.ManagedDeployableContainer.createCommandBuilder(ManagedDeployableContainer.java:60)
	at org.jboss.as.arquillian.container.managed.ManagedDeployableContainer.createCommandBuilder(ManagedDeployableContainer.java:43)
	at org.jboss.as.arquillian.container.CommonManagedDeployableContainer.startInternal(CommonManagedDeployableContainer.java:66)
```

**Comment by scottmarlow** [comment]
The [jarfile.JpaJarFileTestCase](https://ci.wildfly.org/viewLog.html?buildId=328280&buildTypeId=WF_PullRequest_LinuxJdk11#testNameId3860501132310266785) failure needs to be fixed, will take care of that.

**Comment by scottmarlow** [comment]
The one test failure (org.jboss.as.test.integration.domain.mixed eap740.MixedDomain740TestSuite ) doesn't seem related to this pull request.

**Comment by scottmarlow** [comment]
@bstansberry can we remove the legacy Hibernate/EclipeLink versions from boms/legacy-ee/pom.xml?

CC @boris-unckel

**Comment by pferraro** [comment]
@scottmarlow N.B. We can also remove the $legacy.version.org.infinispan property from the parent pom, as well as all infinispan artifacts referencing this version from the legacy-ee bom.  These existed solely to be able to compile the jpa modules that use javax.persistence.  I can also do this in a follow-up PR if you'd prefer.

**Comment by scottmarlow** [comment]
Thanks @pferraro!  I just pushed a change to also remove the Hibernate ORM/EclipseLink + $legacy.version.org.infinispan property from the parent pom.

**Comment by bstansberry** [comment]
@scottmarlow If a dependency listed in boms/legacy-xxx is no longer used anywhere it should be removed. Once a version property isn't used any more it should be removed.

File: jpa/pom.xml [hunk 1]
```diff
@@ -39,31 +39,9 @@
...
-                <module>hibernate5</module>
-                <!-- TODO remove-->
-                <module>hibernate5_3</module>
-                <!-- TODO once moved to jakarta.* move to main module set-->
-                <module>eclipselink</module>
-                <!-- TODO once moved to jakarta.* move to main module set-->
-                <module>openjpa</module>
-            </modules>
-        </profile>
-    </profiles>
```
Comment (bstansberry) [inline jpa/pom.xml:67]: @scottmarlow This profile should be restored and subsystem, spi, eclipselink moved back to it. The others can be removed.

Getting rid of this profile would be done as part of converting the subsystem, spi, eclipselink source code to what's currently used to build ee-9/source-transform/jpa/xxx. Until then this profile allows us to turn off pointless builds of the javax subsystem, spi, eclipselink modules (e.g. to save CI resources).

**Review by bstansberry — APPROVED** [review]

File: jpa/pom.xml (hunk 1 above)
Comment (scottmarlow) [reply jpa/pom.xml:67]: Should we eliminate the `ee-9/source-transform/jpa/eclipselink/pom.xml, ee-9/source-transform/jpa/spi/pom.xml, ee-9/source-transform/jpa/subsystem/pom.xml (removing them from ee-9/source-transform/pom.xml + ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml) as part of this pull request?  As well as switch to jakarta namespace within the wildfly/jpa tree?

Or should we do that separately from this pr?

**Comment by bstansberry** [comment]
Thanks @scottmarlow

**Pull request merged**